src/main/java/com/minecolonies/coremod/datalistener/ResearchListener.java

```diff
@@ -0,0 +1,101 @@
+package com.minecolonies.coremod.datalistener;
+
+import com.google.gson.*;
+import com.minecolonies.api.MinecoloniesAPIProxy;
+import com.minecolonies.api.research.IGlobalResearchTree;
+import com.minecolonies.api.util.Log;
+
+import com.minecolonies.coremod.research.GlobalResearch;
+import net.minecraft.client.resources.JsonReloadListener;
+import net.minecraft.profiler.IProfiler;
+import net.minecraft.resources.IResourceManager;
+import net.minecraft.util.ResourceLocation;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Loader for Json based crafter specific recipes
+ */
+public class ResearchListener extends JsonReloadListener
+{
+    private static final Gson GSON = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
+
+    /**
+     * Set up the core loading, with the directory in the datapack that contains this data
+     * Directory is: <namespace>/researches/<path>
+     */
+    public ResearchListener()
+    {
+        super(GSON, "researches");
+    }
+
+    @Override
+    protected void apply(final Map<ResourceLocation, JsonElement> object, final IResourceManager resourceManagerIn, final IProfiler profilerIn)
+    {
+        Log.getLogger().info("Beginning load of research for University.");
+
+        // First, populate a new map of IGlobalResearches, identified by researchID.
+        // This allows us to figure out root/branch relationships more sanely.
+        final Map<String, GlobalResearch> researchMap = new HashMap<String, GlobalResearch>();
+        for(final Map.Entry<ResourceLocation, JsonElement> entry : object.entrySet())
+        {
+            //Note that we don't actually use the resource folders or file names; those are only for organization purposes.
```

Comment:
We should consider storing the ResourceLocation on each research. This loading code is great for adding researches, but but we need a way to remove researches too, and the best match for that is going to the resource location. 

Please add a mechanism for removing researches.

Reply:
Ok.  I can put that in pretty fast, though I will warn that deregistering a research will need to hide all of its children on the University UI, at least without a significant rewrite there,, and deregistering a researchEffect may cause it to lose progress or reset, so going to put in different methods for each of those two use cases.

Reply:
Yup. There isn't a clean solution for removing a research from an existing world, but for a modpack, removal should allow it to never exist in the world.  As for the children issue, yeah, a child without a parent should either 'disappear' or move to the top.